sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategySuite.scala

```diff
@@ -197,7 +197,7 @@ class FileSourceStrategySuite extends QueryTest with SharedSQLContext with Predi
         "when checking partition values")
     }
     // Only the filters that do not contain the partition column should be pushed down
-    checkDataFilters(Set(IsNotNull("c1"), EqualTo("c1", 1)))
+    checkDataFilters(Set(EqualTo("c1", 1)))
```

Comment:
I believe the new `ConstantPropagation` can remove these `IsNotNull` constraints as there is a stronger `EqualTo`.

Reply:
I am not sure about the reason of this change...where does it come from?

Reply:
The old `ConstantPropagation` replaced an attribute to a constant in `EqualTo` and `EqualNullSafe` expressions only. The new one doesn't make a such limitation.
Since we know that `c1` equals to 1 in this example and I think we can omit this trivial `IsNotNull` filter.

Reply:
Yes, I think it is fine to omit it. Mine was just a question, I am just curious, as I don't really get the reason of this change: I mean I don't understand why this constraint isn't there anymore.

Reply:
I don't see any reason why this limitation (https://github.com/apache/spark/pull/24553/files#diff-a1acb054bc8888376603ef510e6d0ee0L148) was there.